packages/pds/src/services/moderation/index.ts

```diff
@@ -486,7 +500,7 @@ export class ModerationService {
     subject: { did: string } | { uri: AtUri; cid?: CID }
     reportedBy: string
     createdAt?: Date
-  }): Promise<ModerationReportRow> {
+  }): Promise<ModerationReportRowWithSubjectRepo | undefined> {
```

Comment:
I think the contract here would ideally be that it always provides a new report, or it throws an error.  It would be nice if the existence of this new view didn't affect the ability of the report creation method to provide info about the newly-created report.

I notice the joins for `profile` and `did_handle` ended-up getting sprinkled around several places.  One option I could imagine to decouple all these things is to lookup that information in `ModerationView` only when building the `ReportView`, then this method (and some others) wouldn't require any changes.

Reply:
yeah agreed. got rid of the join in almost all places except for the ones where we still need this.